x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -129,40 +136,183 @@ type Props = DraggableProps | NonDraggableProps;
+          {...dragProps}
+          draggable={draggable}
+          reorderableGroup={reorderableGroup}
+          dragging={dragging}
+        />
+      );
+    } else {
+      return <DragInner {...dragProps} draggable={draggable} />;
+    }
+  }
+  if (
+    reorderableGroup &&
+    reorderableGroup.length > 1 &&
+    reorderableGroup?.some((i) => i.id === value.id)
+  ) {
+    return <ReorderableDrop reorderableGroup={reorderableGroup} {...dropProps} />;
+  }
+  return <DropInner {...dropProps} />;
+};
+
+const DragInner = memo(function DragDropInner({
+  dataTestSubj,
+  className,
+  value,
+  children,
+  setDragging,
+  setKeyboardMode,
+  setActiveDropTarget,
+  label = '',
+  keyboardMode,
+  isDragging,
+  activeDropTarget,
+  onDrop,
+  dragType,
+  onDragStart,
+  onDragEnd,
+  extraKeyboardHandler,
+}: DragInnerProps) {
+  const dragStart = (e?: DroppableEvent | React.KeyboardEvent<HTMLButtonElement>) => {
+    // Setting stopPropgagation causes Chrome failures, so
+    // we are manually checking if we've already handled this
+    // in a nested child, and doing nothing if so...
+    if (e && 'dataTransfer' in e && e.dataTransfer.getData('text')) {
+      return;
+    }
+
+    // We only can reach the dragStart method if the element is draggable,
+    // so we know we have DraggableProps if we reach this code.
+    if (e && 'dataTransfer' in e) {
+      e.dataTransfer.setData('text', label);
+    }
+
+    // Chrome causes issues if you try to render from within a
+    // dragStart event, so we drop a setTimeout to avoid that.
+
+    const currentTarget = e?.currentTarget;
+    setTimeout(() => {
+      setDragging(value);
+      if (onDragStart) {
+        onDragStart(currentTarget);
       }
-      isNotDroppable={
-        // If the configuration has provided a droppable flag, but this particular item is not
-        // droppable, then it should be less prominent. Ignores items that are both
-        // draggable and drop targets
-        droppable === false && Boolean(dragging) && value !== dragging
+    });
+  };
+
+  const dragEnd = (e?: DroppableEvent) => {
+    e?.stopPropagation();
+    setDragging(undefined);
+    setActiveDropTarget(undefined);
+    setKeyboardMode(false);
+    if (onDragEnd) {
+      onDragEnd();
+    }
+  };
+
+  const dropToActiveDropTarget = () => {
+    if (isDragging && activeDropTarget?.activeDropTarget) {
+      trackUiEvent('drop_total');
+      if (onDrop) {
+        onDrop(value, activeDropTarget.activeDropTarget);
       }
-    />
+    }
+  };
+
+  return (
+    <div className={className}>
+      <EuiScreenReaderOnly showOnFocus>
+        <button
+          aria-label={label}
+          aria-describedby={`lnsDragDrop-keyboardInstructions`}
+          className="lnsDragDrop__keyboardHandler"
+          data-test-subj="lnsDragDrop-keyboardHandler"
+          onBlur={() => {
+            dragEnd();
+          }}
+          onKeyDown={(e: React.KeyboardEvent<HTMLButtonElement>) => {
```

Comment:
I'd recommend changing the `Enter`/`Space` functionality to the `onClick` callback instead of this custom work in `onKeyDown`. 

Browsers map `onClick` to keyboard events for you for free (and assistive tech can override if they need) without you having to read keyboard events.

Reply:
Unfortunately I cannot make it work this way – the first 'enter' to select the button works, but if I press it again, the 'onClick' handler (that is supposed to deselect the button) doesn't do it.